api/src/test/java/org/openmrs/module/sync2/api/sync/SyncClientTest.java

```diff
@@ -0,0 +1,105 @@
+package org.openmrs.module.sync2.api.sync;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.openmrs.Patient;
+import org.openmrs.PatientIdentifier;
+import org.openmrs.PersonName;
+import org.openmrs.module.sync2.api.Sync2ConfigurationService;
+import org.openmrs.module.sync2.api.impl.Sync2ConfigurationServiceImpl;
+import org.openmrs.module.sync2.client.rest.RestClient;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.Date;
+import java.util.Collections;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.nullValue;
+import static org.hamcrest.core.Is.is;
+import static org.powermock.api.mockito.PowerMockito.*;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({ SyncClient.class })
+public class SyncClientTest {
+
+    private static final String PATIENT_UUID = "patientUuid";
+
+    @Test
+    public void pullDataFromParent_shouldCallRestClient() throws Exception {
+        Patient expectedPatient = createPatient();
+
+        SyncClient resourceManager = new SyncClient();
+        Map<String, String> links = new HashMap<>();
+        links.put("fhir", "openmrs/ws/fhir/Patient/" + PATIENT_UUID);
+        links.put("rest", "openmrs/ws/rest/v1/patient/" + PATIENT_UUID);
+
+        String patientString = "patient";
+        String url = "http://localhost:8080/openmrs/ws/rest/v1/patient/" + PATIENT_UUID;
```

Comment:
Use constants variable for strings:
"fhir",
"openmrs/ws/fhir/Patient/",
"rest",
"openmrs/ws/rest/v1/patient/",
"http://localhost:8080/"